plugins/uefi-dbx/fu-self-test.c

```diff
@@ -206,15 +207,24 @@ fu_self_test_set_up(FuTestFixture *fixture, gconstpointer user_data)
 	gboolean ret;
 	g_autoptr(GError) error = NULL;
 
-	if (tc->with_snapd && fu_self_test_mock_snapd_init(fixture)) {
+	fixture->ctx = fu_context_new();
+
+	/* mocking both running in a snap and detecting fde does not currently make sense */
+	g_assert_false(tc->running_in_snap && tc->snapd_fde_detected);
+
+	if ((tc->running_in_snap || tc->snapd_fde_detected) &&
+	    fu_self_test_mock_snapd_init(fixture)) {
 		fixture->mock_snapd_available = TRUE;
 
-		(void)g_setenv("SNAP", "fwupd", TRUE);
+		if (tc->running_in_snap) {
+			(void)g_setenv("SNAP", "fwupd", TRUE);
+		} else {
+			fu_context_add_flag(fixture->ctx, FU_CONTEXT_FLAG_FDE_SNAPD);
+		}
```

Comment:
in which case, those likely need to be split:
```suggestion
		if (tc->running_in_snap)
			(void)g_setenv("SNAP", "fwupd", TRUE);		
		if (tc->snapd_fde_detected)
			fu_context_add_flag(fixture->ctx, FU_CONTEXT_FLAG_FDE_SNAPD);
	
```